apis/nucleus/src/components/listbox/assets/styling.js

```diff
@@ -1,15 +1,54 @@
 import { getContrastRatio } from '@mui/material/styles';
+import * as namedColors from '@mui/material/colors';
 
 const LIGHT = '#FFF';
 const DARK = '#000';
+const TRANSPARENT = 'rgba(255, 255, 255, 0)';
+
+/**
+ * If needed, converts a named color to a more reliable CSS color format.
+ *
+ * @param {string} c The CSS color.
+ * @returns {string} Converted color or same color as before.
+ * @private
+ * @example
+ *  convertNamedColor('red') => '#FF0000'
+ */
+export function convertNamedColor(c) {
+  let out = c;
+  try {
+    if (c in namedColors) {
+      ({ 500: out } = namedColors[c]); // 500 exposes the standard color
+    } else if (c === 'transparent') {
+      out = TRANSPARENT;
+    }
+  } catch (err) {
+    out = c;
+  }
+  return out;
+}
+
+export const hasEnoughContrast = (desiredTextColor, backgroundColor) => {
+  const desired = convertNamedColor(desiredTextColor);
+  const background = convertNamedColor(backgroundColor);
 
-function getContrastingColor(backgroundColor, desiredTextColor = undefined, dark = DARK, light = LIGHT) {
   const CONTRAST_THRESHOLD = 3.0;
+  let isContrastingEnough;
+  try {
+    isContrastingEnough = getContrastRatio(desired, background) > CONTRAST_THRESHOLD;
+  } catch (err) {
+    // The function throws on unsupported or misspelled colors.
+    // In these cases, simply return true.
```

Comment:
Do you need to do something here?

Reply:
🤔 Yeah, I'm thinking whether to return `true`, `false` or `undefined` in this case. `true` implies the same color as was sent in, so I should probably default to that.